packages/backend/src/api-impl.spec.ts

```diff
@@ -0,0 +1,52 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { expect, test, vi } from 'vitest';
+import examplesCatalog from '../assets/examples.json';
+import type { ExamplesList } from '/@shared/src/models/examples';
+import { BootcApiImpl } from './api-impl';
+import type * as podmanDesktopApi from '@podman-desktop/api';
+
+vi.mock('@podman-desktop/api', async () => {
+  return {
+    window: {
+      showErrorMessage: vi.fn(),
+    },
+    containerEngine: {
+      listImages: vi.fn(),
+    },
+    env: {
+      openExternal: vi.fn(),
+      createTelemetryLogger: vi.fn(),
+    },
+  };
+});
+
+test('getExamples should return examplesCatalog', async () => {
+  const extensionContextMock = {
+    storagePath: '/fake-path',
+  } as unknown as podmanDesktopApi.ExtensionContext;
+
+  const webviewMock = {} as podmanDesktopApi.Webview;
+
+  const bootcApi = new BootcApiImpl(extensionContextMock, webviewMock);
+
+  // When running get Examples we should return the examplesCatalog (it's exported)
+  const result = await bootcApi.getExamples();
+  expect(result).toEqual(examplesCatalog as ExamplesList);
```

Comment:
This is a really simple test. Could you at least check examples length and categories length is > 0 too?